admins/views.py

```diff
@@ -416,8 +433,278 @@ def delete_timetable_entry(request, entry_id):
+    """
+    try:
+        student = Student.objects.get(USN=student_id)
+    except Student.DoesNotExist:
+        messages.error(request, _('The student does not exist!'))
+        return redirect('class_list')
+
+    if request.method == 'POST':
+        # Sử dụng EditStudentForm thay vì AddStudentForm
+        form = EditStudentForm(request.POST, instance=student)
+
+        if form.is_valid():
+            try:
+                with transaction.atomic():
+                    # Cập nhật User model
+                    user = student.user
+
+                    # Cập nhật username nếu khác
+                    new_username = form.cleaned_data.get('username')
+                    if new_username and new_username != user.username:
+                        user.username = new_username
+
+                    # Cập nhật email nếu khác
+                    new_email = form.cleaned_data.get('email')
+                    if new_email and new_email != user.email:
+                        user.email = new_email
+
+                    # Cập nhật password nếu được cung cấp
+                    new_password = form.cleaned_data.get('password')
+                    if new_password and new_password.strip():
+                        user.set_password(new_password)
+
+                    # Cập nhật first_name và last_name từ name
+                    new_name = form.cleaned_data.get('name')
+                    if new_name:
+                        name_parts = new_name.split()
+                        user.first_name = name_parts[0] if name_parts else ''
+                        user.last_name = ' '.join(name_parts[1:]) if len(
+                            name_parts) > 1 else ''
+
+                    user.save()
+
+                    # Cập nhật Student model (form.save() sẽ tự động update vì có instance)
+                    form.save()
+
+                    messages.success(request, _(
+                        'Student "{}" has been updated successfully!').format(student.name))
+                    return redirect('edit_class', class_id=student.class_id.id)
+
+            except Exception as e:
+                messages.error(request, _(
+                    'Error updating student: {}').format(str(e)))
+        else:
+            # Hiển thị lỗi validation
+            for field, errors in form.errors.items():
+                for error in errors:
+                    messages.error(request, f"{field}: {error}")
+    else:
+        # GET request - hiển thị form với dữ liệu hiện tại
+        form = EditStudentForm(instance=student)
+
+    context = {
+        'form': form,
+        'student': student,
+        'admin_user': request.user,
+        'title': _('Edit Student {}').format(student.name),
+        'submit_text': _('Update Student'),
+    }
+    return render(request, 'admins/edit_student.html', context)
+
+
+@login_required
+@permission_required('students.delete_student', raise_exception=True)
+def delete_student(request, student_id):
```

Comment:
Thao tác xóa student cũng chú ý check logic liên quan, vd học sinh đã tham gia học, có kết quả môn học thì chỉ nên Deactive tài khoản (vd: học sinh chuyển trường, nghỉ học)
còn nếu hs chưa tham gia học thì có thể cho phép xóa hoàn toàn